README.md

```diff
@@ -1048,6 +1049,12 @@ if (name == "Hello") {
 }
 ```
 
+## Use of self
+
+Don't use `self`, unless your code can't compile without it.
+
+Use `self` only when required by the compiler (in @escaping closures, or in initializers to disambiguate properties from arguments).
```

Comment:
```suggestion
Use `self` only when required by the compiler (in `@escaping` closures, or in initializers to disambiguate properties from arguments).
```

In the case of the initializers though is worth making an exception:

Consider the following:

```swift
init(price: Int, color: UIColor) {
  self.price = price
  theColor = color
}
```

It doesn't read as homogeneous as the version with `self.` even if it's not necessary to disambiguate.

```swift
init(price: Int, color: UIColor) {
  self.price = price
  self.theColor = color
}
```

The exception to the exception (🙃) would be `init(from decoder:)` as we discussed, where non of the properties are ambiguous so no `self.` is required.